jib-core/src/main/java/com/google/cloud/tools/jib/docker/DockerContextGenerator.java

```diff
@@ -119,35 +126,36 @@ String makeDockerfile() throws IOException {
             "@@DEPENDENCIES_PATH_ON_IMAGE@@", sourceFilesConfiguration.getDependenciesPathOnImage())
         .replace("@@RESOURCES_PATH_ON_IMAGE@@", sourceFilesConfiguration.getResourcesPathOnImage())
         .replace("@@CLASSES_PATH_ON_IMAGE@@", sourceFilesConfiguration.getClassesPathOnImage())
-        .replace("@@ENTRYPOINT@@", getEntrypoint());
+        .replace(
+            "@@ENTRYPOINT@@",
+            makeDockerList(
+                EntrypointBuilder.makeEntrypoint(sourceFilesConfiguration, jvmFlags, mainClass)))
+        .replace("@@CMD@@", makeDockerList(javaArguments));
   }
 
   /**
-   * Gets the Dockerfile ENTRYPOINT in exec-form.
+   * Formats a list for the Dockerfile's ENTRYPOINT or CMD.
    *
    * @see <a
    *     href="https://docs.docker.com/engine/reference/builder/#exec-form-entrypoint-example">https://docs.docker.com/engine/reference/builder/#exec-form-entrypoint-example</a>
    */
   @VisibleForTesting
-  String getEntrypoint() {
-    List<String> entrypoint =
-        EntrypointBuilder.makeEntrypoint(sourceFilesConfiguration, jvmFlags, mainClass);
-
-    StringBuilder entrypointString = new StringBuilder("[");
+  String makeDockerList(List<String> items) {
```

Comment:
Looks like this can be a static method now. Can make this more descriptively named like `joinAsJsonArray`, which is the form that Dockerfile takes.